### PR TITLE
docs: add maintainers release guide

### LINKED ISSUE #4042: docs: document the release process
Create central documentation on how to release Invenio. This documentation should at least answer to:
- [ ] How to release Invenio framework?
- [ ] How to release Invenio modules?
- [ ] How to manually release a module?
- [ ] How to merge and release to maintenance?

Note: pay extra attention to dependency management.

For example: https://digital-repositories.web.cern.ch/digital-repositories/reana/releasing/ (to be migrated to http://docs.reana.io/development/releasing/)

### PR DESCRIPTION
**Pull request by ntarocco**
- closes #4042

### CODE REVIEW

File: docs/community/maintainers-guide/releasing.rst [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+=========
+
+In general, you have two types of releases:
+
+- **Major/Minor-level release** - Often these releases add significant new
+  features and needs to go through extensive testing.
+- **Patch-level release** - Often these releases fix security issues or
+  important bugs.
+
+For major/minor-level releases, you'll often be working only with the
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:18]: Minor: just cosmetic. When reading I feel that "often" is used a lot, we could exchange it with synonyms at least in this paragraph, e.g.

```suggestion
For major/minor-level releases, you'll mostly be working only with the
```

File: docs/community/maintainers-guide/releasing.rst [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    This file is part of Invenio.
+    Copyright (C) 2020 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+Releasing
+=========
+
+In general, you have two types of releases:
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:11]: Minor: I would make it shorter, what do you think?
```suggestion
The most common releases are:
```

File: docs/community/maintainers-guide/releasing.rst [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+- **Major/Minor-level release** - Often these releases add significant new
+  features and needs to go through extensive testing.
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:14]:
```suggestion
  features and need to go through extensive testing.
```

File: docs/community/maintainers-guide/releasing.rst [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+For major/minor-level releases, you'll often be working only with the
+``master``-branch and only make a single PyPI release. For patch-level
+releases you'll often be working with multiple maintenance branches and make
+several PyPI releases for all currently maintained versions of Invenio.
+
+Depending on the package that you are releasing, you will have to follow
+a different procedure. You might have to release:
+
+- one of the Python or JavaScript/React modules (happening rather often)
+- the Invenio framework, the ``invenio`` package (happening less often)
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:27]: Maybe for more clarity the "`invenio` package" can be a link to the page in PyPI?

File: docs/community/maintainers-guide/releasing.rst [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+For major/minor-level releases, you'll often be working only with the
+``master``-branch and only make a single PyPI release. For patch-level
+releases you'll often be working with multiple maintenance branches and make
+several PyPI releases for all currently maintained versions of Invenio.
+
+Depending on the package that you are releasing, you will have to follow
+a different procedure. You might have to release:
+
+- one of the Python or JavaScript/React modules (happening rather often)
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:26]: Minor: I think that examples are always good so I would add links to released modules. 

```suggestion
- one of the Python (e.g. `Invenio-DB <https://pypi.org/project/invenio-db/>`_) or JavaScript/React (e.g. `React-SearchKit <https://www.npmjs.com/package/react-searchkit>`_) modules (happening rather often)
```

File: docs/community/maintainers-guide/releasing.rst [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+1. Check that all relevant PRs have been merged. For the ``invenio`` package,
+   check that all relevant Invenio modules have been released.
+2. Fetch latest changes:
+    - ``git fetch inveniosoftware``
+3. Checkout maintenance or master branch:
+    - ``git checkout maint-<major>.<minor>``
+4. Merge or reset to latest changes:
+    - ``git merge --ff-only inveniosoftware/maint-<major>.<minor>``
+    - or
+    - ``git reset --hard inveniosoftware/maint-<major>.<minor>``
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:43]: Minor: personally I would add a `git reset --hard` so easily copy-pastable, I would rather say between parenthesis "if you cannot fast-forward merge make sure you save your local changes in a different branch if you need to, and reset to `inveniosoftware/maint-<major>.<minor>`" or shorter message.

File: docs/community/maintainers-guide/releasing.rst [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+
+While above is not strictly necessary, it avoids situations where you local
+and GitHub branches have diverged.
+
+Patch-level release
+-------------------
+Most of the time, patch-level releases are not necessary for the main
+``invenio`` package, since by patch-level releases of the underlying
+modules are automatically picked up: the dependencies upper bounds are
+normally allowing the installation of patch-level releases.
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:53]: Minor: Maybe we can add a link to illustrate this? Like [this one](https://github.com/inveniosoftware/invenio/blob/0df2bdedbf3b40ea970eed1883f465e24e26d100/setup.py#L24-L83), I am not sure since this is a permalink to current master and things may change.. what do you think?

File: docs/community/maintainers-guide/releasing.rst [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+
+Most of the Invenio modules have maintenance branches for previous
+major/minor releases. They are normally named ``maint-<version>``.
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:56]:
```suggestion
major/minor releases. They are normally named ``maint-<version>``, e.g. `Invenio-Records-Rest maint-1.6 <https://github.com/inveniosoftware/invenio-records-rest/tree/maint-1.6>_`.
```

File: docs/community/maintainers-guide/releasing.rst [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+compatibility with the current released ``invenio`` version, by checking
+the upper bounds defined for this module in the ``invenio`` releases.
+
+The most probable scenario is that for a given ``invenio`` release,
+e.g. ``3.3``, the upper bound for an Invenio module ``foo``
+(with current latest release ``1.2.6``) is set to ``invenio-foo<1.3.0``.
+A new minor release ``1.3.0`` of this module will not affect
+``invenio 3.3``.
+
+If this is not the case, you will have to make sure that the new release
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:74]: I think this paragraph deserves a special formatting warning or similar if warning it too much https://sublime-and-sphinx-guide.readthedocs.io/en/latest/notes_warnings.html, what do you think?

File: docs/community/maintainers-guide/releasing.rst [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+
+- **Major/Minor-level release** - Often these releases add significant new
+  features and needs to go through extensive testing.
+- **Patch-level release** - Often these releases fix security issues or
+  important bugs.
+
+For major/minor-level releases, you'll often be working only with the
+``master``-branch and only make a single PyPI release. For patch-level
+releases you'll often be working with multiple maintenance branches and make
+several PyPI releases for all currently maintained versions of Invenio.
```
Comment (ppanero) [inline docs/community/maintainers-guide/releasing.rst:21]: Minor: might be worth mention that this is what's called *a backport release*?

File: docs/community/maintainers-guide/releasing.rst [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+      highlighting backward incompatible ones, and that the release date
+      is correct. Pay attention how you write, keep consistency with previous
+      sentences forms.
+4. Bump version in ``version.py`` (follow semantic versioning).
+5. Create a release commit (message pattern:
+   ``release: v<major>.<minor>.<patch>``).
+6. Create tag and push release:
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware master v<major>.<minor>.<patch>``
+7. Regularly check ``Travis`` and ``PyPI`` to ensure the release went through.
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:99]: What about saying something along the lines of:
```
Confirm that the Travis build run successfully and that the PyPI release is completed, e.g. https://travis-ci.org/github/inveniosoftware/invenio-records-rest/builds/684646338 and https://pypi.org/project/invenio-records-rest/1.6.5/
```

File: docs/community/maintainers-guide/releasing.rst (hunk 5 above)
Comment (ppanero) [inline docs/community/maintainers-guide/releasing.rst:26]: Minor: The other enumerations in this document are capitalised and end in dot.

```suggestion
- One of the Python or JavaScript/React modules (happening rather often).
```

File: docs/community/maintainers-guide/releasing.rst (hunk 4 above)
Comment (ppanero) [inline docs/community/maintainers-guide/releasing.rst:27]: Same:
```suggestion
- The Invenio framework, the ``invenio`` package (happening less often).
```

File: docs/community/maintainers-guide/releasing.rst [hunk 12]
```diff
@@ -0,0 +1,224 @@
...
+
+Before you start
+----------------
+Prior to preparing a new Invenio release, please ensure you have the latest
+changes in your local environment for the maintenance branches and ``master``:
+
+1. Check that all relevant PRs have been merged. For the ``invenio`` package,
+   check that all relevant Invenio modules have been released.
+2. Fetch latest changes:
+    - ``git fetch inveniosoftware``
```
Comment (ppanero) [inline docs/community/maintainers-guide/releasing.rst:37]: Minor: I think a maintainer should be more than aware of this... But `inveniosoftware` is a remote, might have another name (i.e. in my case is `upstream` to be compatible with `hub`). Still as said, not sure is worth mentioning?

File: docs/community/maintainers-guide/releasing.rst [hunk 13]
```diff
@@ -0,0 +1,224 @@
...
+4. Bump version in ``package.json`` (follow semantic versioning).
+5. Regenerate the ``package-lock.json``:
+    - ``rm package-lock.json``
+    - ``npm install``
+6. Create a release commit (message pattern:
+   ``release: v<major>.<minor>.<patch>``).
+7. Create tag and push release:
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware master v<major>.<minor>.<patch>``
+8. Regularly check ``Travis`` and ``NpmJS`` to ensure the release went through.
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:119]: If https://github.com/inveniosoftware/invenio/pull/4044/files#r474518553 makes sense I would do the same here but adapted to JS.

File: docs/community/maintainers-guide/releasing.rst [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+When you need to create a new release for a maintained previous release,
+you will have to do the extra step of backporting the changes to the
+maintenance branch.
+
+After the procedure described above, you will have to:
+
+1. Checkout the ``maint`` branch and create a new release branch:
+    - ``git checkout maint-<major>.<minor>``
+    - ``git checkout -b rel-v<major>.<minor>.<patch>``
+2. Cherry-pick the commits that you need to backport, resolve any conflict.
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:132]: What do you think about adding a link to cherry-pick docs? https://git-scm.com/docs/git-cherry-pick

File: docs/community/maintainers-guide/releasing.rst [hunk 15]
```diff
@@ -0,0 +1,224 @@
...
+   (yes, cherry-pick :))
+3. Run tests.
+4. Createa new commit with updated ``changes`` file and bumped version,
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:135]: What do you mean with as described above?

File: docs/community/maintainers-guide/releasing.rst [hunk 16]
```diff
@@ -0,0 +1,224 @@
...
+**Python module**
+
+1. Make sure that all relevant PRs have been merged to ``master``.
+2. Make sure that the build on ``master`` branch is passing.
+3. Prepare release notes in
+    - Update the ``CHANGE.rst`` file, making sure that you describe changes,
+      highlighting backward incompatible ones, and that the release date
+      is correct. Pay attention how you write, keep consistency with previous
+      sentences forms.
+4. Bump version in ``version.py`` (follow semantic versioning).
```
Comment (ppanero) [inline docs/community/maintainers-guide/releasing.rst:93]: Moderate: Would it be worth adding a step here like:

5. If the module is changing from alpha/beta to official release remember to modify the classifiers in the `setup.py` according to <https://pypi.org/classifiers/>.

File: docs/community/maintainers-guide/releasing.rst [hunk 17]
```diff
@@ -0,0 +1,224 @@
...
+    - ``git checkout -b rel-v<major>.<minor>.<patch>``
+2. Cherry-pick the commits that you need to backport, resolve any conflict.
+   (yes, cherry-pick :))
+3. Run tests.
+4. Createa new commit with updated ``changes`` file and bumped version,
+   as described above.
+5. Issue a pull request against the **maintenance branch** (
+   ``maint-<major>.<minor>``).
+6. If Travis fails:
+    - Fix issue and ensure head commit is the release commit (i.e. rebase if
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:140]: I see you highlight important info in bold right? I would put in bold the part "ensure head commit is the release commit"

File: docs/community/maintainers-guide/releasing.rst (hunk 15 above)
Comment (ppanero) [inline docs/community/maintainers-guide/releasing.rst:135]:
```suggestion
4. Create a new commit with updated ``changes`` file and bumped version,
```

Comment (ppanero) [inline docs/community/maintainers-guide/releasing.rst:135]: Minor: In other cases it is mentioned as `CHANGE.md`. So I would go for that or *changes* without quotes. I think quotes should be used for specific -exact- names.

File: docs/community/maintainers-guide/releasing.rst [hunk 18]
```diff
@@ -0,0 +1,224 @@
...
+
+Invenio release
+---------------
+
+The pre-requirement necessary to release Invenio is that all the features in
+the various Invenio modules needed for the release have been merged and
+released. Then:
+
+1. Update the ``setup.py``:
+    - review all modules lower and upper bounds and adjust them as needed.
```
Comment (ppanero) [inline docs/community/maintainers-guide/releasing.rst:155]:
```suggestion
    - Review all modules lower and upper bounds and adjust them as needed.
```

File: docs/community/maintainers-guide/releasing.rst [hunk 19]
```diff
@@ -0,0 +1,224 @@
...
+3. Run tests.
+4. Createa new commit with updated ``changes`` file and bumped version,
+   as described above.
+5. Issue a pull request against the **maintenance branch** (
+   ``maint-<major>.<minor>``).
+6. If Travis fails:
+    - Fix issue and ensure head commit is the release commit (i.e. rebase if
+      necessary).
+7. Merge, tag and push release:
+    - ``git merge --ff-only rel-v<major>.<minor>.<patch>``
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:143]: We are missing this bullet before, right?:
```console
- ``git checkout maint-<major>.<minor>``
```

File: docs/community/maintainers-guide/releasing.rst [hunk 20]
```diff
@@ -0,0 +1,224 @@
...
+---------------
+
+The pre-requirement necessary to release Invenio is that all the features in
+the various Invenio modules needed for the release have been merged and
+released. Then:
+
+1. Update the ``setup.py``:
+    - review all modules lower and upper bounds and adjust them as needed.
+2. Documentation:
+    - review documentation and make sure new features or breaking changes are
```
Comment (ppanero) [inline docs/community/maintainers-guide/releasing.rst:157]:
```suggestion
    - Review documentation and make sure new features or breaking changes are
```

File: docs/community/maintainers-guide/releasing.rst [hunk 21]
```diff
@@ -0,0 +1,224 @@
...
+   as described above.
+5. Issue a pull request against the **maintenance branch** (
+   ``maint-<major>.<minor>``).
+6. If Travis fails:
+    - Fix issue and ensure head commit is the release commit (i.e. rebase if
+      necessary).
+7. Merge, tag and push release:
+    - ``git merge --ff-only rel-v<major>.<minor>.<patch>``
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware maint-<major>.<minor> v<major>.<minor>.<patch>``
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:145]: Do you have any backport at reach? We could add a link to it as an example.

File: docs/community/maintainers-guide/releasing.rst [hunk 22]
```diff
@@ -0,0 +1,224 @@
...
+    - Edit release notes.
+    - Include the new release notes into ``docs/releases/index.rst``.
+    - Check the "Maintenance policy", e.g. is the version correct?
+      (`example <https://github.com/inveniosoftware/invenio/commit/edb863d2f5228fb158c090a69c2db7c3385b6ba3>`_).
+4. Create a release commit (message pattern:
+   ``release: v<major>.<minor>.<patch>``).
+5. Create tag and push release:
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware master v<major>.<minor>.<patch>``
+6. Regularly check ``Travis`` and ``PyPI`` to ensure the release went through.
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:171]: I would rephrase as in https://github.com/inveniosoftware/invenio/pull/4044/files#r474518553

File: docs/community/maintainers-guide/releasing.rst [hunk 23]
```diff
@@ -0,0 +1,224 @@
...
+
+You can manually reproduce the publishing process done by ``Travis`` by doing:
+
+1. Activate your virtualenv for the package that you want to release.
+2. Generate the different distributions:
+    - ``python setup.py compile_catalog sdist bdist_wheel``
+3. Install the tool to upload releases to PyPI:
+    - ``pip install twine wheel``
+4. Publish:
+    - ``twine upload --repository <repo-name> dist/*``. The command will
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:189]: Thinking of newcomers to the role: maybe we could point out that they can test the release in https://test.pypi.org/ (see [docs](https://twine.readthedocs.io/en/latest/#using-twine)).

Comment (diegodelemos) [reply docs/community/maintainers-guide/releasing.rst:189]: - I think that the `<repo-name>` placeholder in `twine upload --repository <repo-name> dist/*` can be confused with the Invenio module repository. 
- We can add a link to the architects: https://github.com/orgs/inveniosoftware/teams/architects
```console
4. Publish:
    - ``twine upload dist/*``. The command will ask for username and password. `Invenio architects <https://github.com/orgs/inveniosoftware/teams/architects>`_ should have the credentials.
.. note::
   If it is your first time releasing in PyPI, or you are not sure if the release is correct, you can test it in https://test.pypi.org/ see the `documentation 
<https://twine.readthedocs.io/en/latest/#using-twine>`_.
```

File: docs/community/maintainers-guide/releasing.rst [hunk 24]
```diff
@@ -0,0 +1,224 @@
...
+**NpmJS**
+
+Manual release on ``NpmJS`` is not only needed in case of failure, but also
+when creating a totally new package, never released yet.
+
+1. Make sure that you have an updated version of the
+   `npm <https://www.npmjs.com/get-npm>`_ client installed in your machine.
+2. Login on ``NpmJS``:
+    - ``npm login``: you will need ``inveniosoftware`` username and password.
+      Architects should have the credentials.
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:201]: Link to architects?

File: docs/community/maintainers-guide/releasing.rst [hunk 25]
```diff
@@ -0,0 +1,224 @@
...
+3. Optionally, verify the package before releasing:
+    - ``npm pack``: this will create the final archive that will be
+      published on ``NpmJS`` in case you want to check its content.
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:204]: Just for the harmony of the docs: why don't we add this step to the PyPI release? the content of the release can as well be consulted in `sdist` after running `python setup.py sdist`.

File: docs/community/maintainers-guide/releasing.rst [hunk 26]
```diff
@@ -0,0 +1,224 @@
...
+      Run ``npm publish --access public --dry-run`` to double check that
+      everything is ok, then ``npm publish --access public``.
+
+
+Announcing release
+==================
+
+You should announce a new Invenio release in the following channels:
+
+- Invenio-Talk Announcement
```
Comment (diegodelemos) [inline docs/community/maintainers-guide/releasing.rst:222]: What do you think of adding links to examples of all of these communications? so we save adding a template here

File: docs/community/maintainers-guide/releasing.rst (hunk 10 above)
Comment (lnielsen) [reply docs/community/maintainers-guide/releasing.rst:21]: It's not really a backport, since a backport is a new feature added in a new release, that is being ported to work on an older release. We don't do that. We make bug fixes on all supported releases.

File: docs/community/maintainers-guide/releasing.rst [hunk 27]
```diff
@@ -0,0 +1,224 @@
...
+
+- one of the Python or JavaScript/React modules (happening rather often)
+- the Invenio framework, the ``invenio`` package (happening less often)
+
+Before you start
+----------------
+Prior to preparing a new Invenio release, please ensure you have the latest
+changes in your local environment for the maintenance branches and ``master``:
+
+1. Check that all relevant PRs have been merged. For the ``invenio`` package,
```
Comment (lnielsen) [inline docs/community/maintainers-guide/releasing.rst:34]: Styling for bullet one is a bit off:
Screenshot 2020-08-26 at 14 30 43

File: docs/community/maintainers-guide/releasing.rst (hunk 12 above)
Comment (lnielsen) [reply docs/community/maintainers-guide/releasing.rst:37]: I'm using hub, and I get ``inveniosoftware`` as remote :-) That said, I would include this in a different section on developer environment setup and simply link there.

File: docs/community/maintainers-guide/releasing.rst [hunk 28]
```diff
@@ -0,0 +1,224 @@
...
+2. Make sure that the build on ``master`` branch is passing.
+3. Prepare release notes in
+    - Update the ``CHANGE.rst`` file, making sure that you describe changes,
+      highlighting backward incompatible ones, and that the release date
+      is correct. Pay attention how you write, keep consistency with previous
+      sentences forms.
+4. Bump version in ``version.py`` (follow semantic versioning).
+5. Create a release commit (message pattern:
+   ``release: v<major>.<minor>.<patch>``).
+6. Create tag and push release:
```
Comment (lnielsen) [inline docs/community/maintainers-guide/releasing.rst:96]: Normally we open a pull request first and see it pass.

File: docs/community/maintainers-guide/releasing.rst [hunk 29]
```diff
@@ -0,0 +1,224 @@
...
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware master v<major>.<minor>.<patch>``
+7. Regularly check ``Travis`` and ``PyPI`` to ensure the release went through.
+
+**JavaScript/React module**
+
+1. Make sure that all relevant PRs have been merged to ``master``.
```
Comment (lnielsen) [inline docs/community/maintainers-guide/releasing.rst:103]: I would put the common points, into a common release checklist.
1) The checklist requires more thought (e.g. what do I put in CHANGE.rst), has all been merged etc., where as
2) the mechanics of releasing is a recipe.

File: docs/community/maintainers-guide/releasing.rst [hunk 30]
```diff
@@ -0,0 +1,224 @@
...
+    - ``rm package-lock.json``
+    - ``npm install``
+6. Create a release commit (message pattern:
+   ``release: v<major>.<minor>.<patch>``).
+7. Create tag and push release:
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware master v<major>.<minor>.<patch>``
+8. Regularly check ``Travis`` and ``NpmJS`` to ensure the release went through.
+
+**Backport changes to a maint release**
```
Comment (lnielsen) [inline docs/community/maintainers-guide/releasing.rst:121]: Comment: I wouldn't call it backport. It can happen in both directions. First fix in maint, then merged to master - or fix in master merge to maint.

Comment (lnielsen) [reply docs/community/maintainers-guide/releasing.rst:121]: Perhaps something like "Patch-level releases: moving between branches" or "Releasing bug fixes on multiple branches".

File: docs/community/maintainers-guide/releasing.rst (hunk 14 above)
Comment (lnielsen) [reply docs/community/maintainers-guide/releasing.rst:132]: I would follow the style:

``git cherry-pick <commit id>``

File: docs/community/maintainers-guide/releasing.rst [hunk 31]
```diff
@@ -0,0 +1,224 @@
...
+When the process of releasing fails for some reasons, you might want to
+manually publish the new version of a package.
+
+**PyPI**
+
+You can manually reproduce the publishing process done by ``Travis`` by doing:
+
+1. Activate your virtualenv for the package that you want to release.
+2. Generate the different distributions:
+    - ``python setup.py compile_catalog sdist bdist_wheel``
```
Comment (lnielsen) [inline docs/community/maintainers-guide/releasing.rst:185]: compile_catalog might be optional (see travis.yml)

File: docs/community/maintainers-guide/releasing.rst (hunk 19 above)
Comment (ppanero) [reply docs/community/maintainers-guide/releasing.rst:143]: It should have been done in step 1 of this instructions.

File: docs/community/maintainers-guide/releasing.rst (hunk 27 above)
Comment (ppanero) [reply docs/community/maintainers-guide/releasing.rst:34]: The identation in the code seems correct. If put one tab less, then the enumerated list is failing (ends and restarts after...). Same if trying 2 spaces instead of a full tab. e.g

```
/Users/ppanero/Workspace/inveniosw/invenio/docs/community/contributing/maintainers-guide/releasing.rst:39:Enumerated list ends without a blank line; unexpected unindent.
```

File: docs/community/maintainers-guide/releasing.rst (hunk 6 above)
Comment (ppanero) [reply docs/community/maintainers-guide/releasing.rst:45]: Im on team Nico on this one. One possible problem I see with having a simple `git reset --hard` copy-pastable is that if not properly configured it can actually reset all branches, and it is sort of a destructive action. In addition the rest of the docs follo the <> in terms of "fill with your concrete case"

File: docs/community/maintainers-guide/releasing.rst (hunk 7 above)
Comment (ppanero) [reply docs/community/maintainers-guide/releasing.rst:55]: I think point to master is like point to line... it might change a lot in the future. If needed I would add one example. e.g.:

```python
        'invenio-records-rest>=1.7.1,<1.8.0',
        'invenio-records-ui>=1.2.0a1,<1.3.0',
```

File: docs/community/maintainers-guide/releasing.rst (hunk 11 above)
Comment (ppanero) [reply docs/community/maintainers-guide/releasing.rst]: personally I think we are adding too many links and it might be difficult to follow (e.g. get lost opening tabs). That said, I dont have a strong opinion against that, so it can be added. WDYT?

File: docs/community/maintainers-guide/releasing.rst (hunk 13 above)
Comment (ppanero) [reply docs/community/maintainers-guide/releasing.rst:125]: Same answer as above :)

File: docs/community/maintainers-guide/releasing.rst (hunk 21 above)
Comment (ppanero) [reply docs/community/maintainers-guide/releasing.rst:151]: hmm by backport you mean github link? The commands will not be illustrated there, just a list of commits, no? Otherwise, I did not fully understood this.

File: docs/community/maintainers-guide/releasing.rst (hunk 22 above)
Comment (ppanero) [reply docs/community/maintainers-guide/releasing.rst:177]: same, might need discussion IRL

File: docs/community/maintainers-guide/releasing.rst (hunk 26 above)
Comment (ppanero) [reply docs/community/maintainers-guide/releasing.rst:240]: We can add pointers, templates do not exists per se (IIRC). E.g. the announcement might differ a lot depending on the release. What about:

- https://inveniosoftware.org/blog/invenio-v33-released/
- https://invenio-talk.web.cern.ch/t/invenio-v3-2-released/62/3
- https://discordapp.com/channels/692989811736182844/704628216857362473/740206015957434439

There is only announcement of RDM ones in discord... Maybe we can wait till v3.4 is out and put all together the same?

File: docs/community/maintainers-guide/releasing.rst (hunk 28 above)
Comment (ppanero) [reply docs/community/maintainers-guide/releasing.rst]: you mean add the step above? As travis is set the push will still go by travis... I dont fully understand this.

File: docs/community/maintainers-guide/releasing.rst (hunk 29 above)
Comment (ppanero) [reply docs/community/maintainers-guide/releasing.rst:109]: IMO is easier to follow steps all together, instead of jumping in sections for specific cases. However, it's true that it might have a higher impact in maintainance.

File: docs/community/maintainers-guide/releasing.rst (hunk 30 above)
Comment (ppanero) [reply docs/community/maintainers-guide/releasing.rst:127]: I have no strong preference, what is the consensus on this one?

File: docs/community/maintainers-guide/releasing.rst (hunk 6 above)
Comment (diegodelemos) [reply docs/community/maintainers-guide/releasing.rst]: That is precisely what I originally meant if it wasn't because of a very tricky typo (see that if you read it now it makes more sense the whole sentence since I say "just say it in words and let the person to manage as they know"):
```diff
- Minor: personally I would add a git reset --hard so easily copy-pastable, I would rather say between parenthesis "if you cannot fast-forward merge make sure you save your local changes in a different branch if you need to, and reset to inveniosoftware/maint-<major>.<minor>" or shorter message.
+ Minor: personally I wouldn't add a git reset --hard so easily copy-pastable, I would rather say between parenthesis "if you cannot fast-forward merge make sure you save your local changes in a different branch if you need to, and reset to inveniosoftware/maint-<major>.<minor>" or shorter message.
```

**Comment by diegodelemos** [comment]
For speed purposes let's solve all minor comments by ignoring them as the new contributing docs state and get this merged.

For the bullet points, if we don't have a solution right now, I would say, let's just create an issue about it and move forward.

File: docs/community/maintainers-guide/releasing.rst (hunk 6 above)
Comment (ntarocco) [reply docs/community/maintainers-guide/releasing.rst]: Fixed

File: docs/community/maintainers-guide/releasing.rst (hunk 7 above)
Comment (ntarocco) [reply docs/community/maintainers-guide/releasing.rst]: The link to the commit is persistent, a permalink, unless you push force and re-write the commit. I will add the example.

File: docs/community/maintainers-guide/releasing.rst (hunk 11 above)
Comment (ntarocco) [reply docs/community/maintainers-guide/releasing.rst]: Kind of agree on this one, maintainer should know how to find Travis or PyPI pages?

File: docs/community/maintainers-guide/releasing.rst (hunk 25 above)
Comment (ntarocco) [reply docs/community/maintainers-guide/releasing.rst]: I am not sure I have understood this comment, sorry :(

File: docs/community/maintainers-guide/releasing.rst (hunk 26 above)
Comment (ntarocco) [reply docs/community/maintainers-guide/releasing.rst]: Links added but Discord!

File: docs/community/maintainers-guide/releasing.rst (hunk 27 above)
Comment (ntarocco) [reply docs/community/maintainers-guide/releasing.rst]: I have changed, see [here](https://stackoverflow.com/a/5553270/675178)... because we ❤️ Sphinx, right? :)

File: docs/community/maintainers-guide/releasing.rst (hunk 29 above)
Comment (ntarocco) [reply docs/community/maintainers-guide/releasing.rst]: Honestly, I started to look into that, checking other products too, but I was not sure where to stop. Let's start like this and we improve it.

File: docs/community/maintainers-guide/releasing.rst (hunk 30 above)
Comment (ntarocco) [reply docs/community/maintainers-guide/releasing.rst]: I went for `Patch-level releases on multiple maint branches`

**Review by Aeinst-ai — COMMENTED** [review]
https://github.com/CocoaPods/CocoaPods/wiki/Communication-&-Design-Rules

**Comment by Aeinst-ai** [comment]
Building stronger communities with clarification staying on top of your game plan  by communicating revising reviewing  promoting a respectful new normal masked COVID-19 Pandemic environmental topics that help produce positive and productive discussions respectively.
